mmrazor/models/mutables/mutable_channel/groups/channel_group.py

```diff
@@ -0,0 +1,730 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+"""This module defines ChannelGroup with related modules.
+
+PruneNode                               Channel
+                ------------------->
+PruneGraph      Graph2ChannelGroups     ChannelGroup
+
+PruneNode and PruneGraph are used to record the computation graph of a model.
+A Channel records a slice of the input or output channels of a module.
+A ChannelGroup collects all Channels with channel-dependency.
+Graph2ChannelGroups is used to parse a PruneGraph and get ChannelGroups
+"""
+
+import copy
+from typing import Any, Dict, List, Tuple, TypeVar, Union
+
+import torch.nn as nn
+from torch.nn import Module
+
+from mmrazor.models.architectures.dynamic_ops.bricks.dynamic_mixins import \
```

Comment:
Now, it's  difficult to import from mmrazor.models.architectures, as old ops are also used in some algorithm, like autoslim  and slimmable. This may be changed after merging refactored autoslim and slimmable.